migration_test/migration.test.ts

```diff
@@ -0,0 +1,637 @@
+        to: randomUser.address,
+        value: parseEther('0.1'),
+      })
+    ).wait()
+
+    const depositRes = await ethBridger.deposit({
+      l1Signer: randomUser,
+      l2Provider: this.testState.core.l2Signer.provider!,
+      amount: amountToSend,
+    })
+    await depositRes.wait()
+
+    this.testState.ethXChainDeposit.user = randomUser
+    this.testState.ethXChainDeposit.balance = amountToSend
+  }
+
+  public override async finalise(): Promise<void> {
+    const l1User = this.testState.ethXChainDeposit.user!.connect(
+      this.testState.core.l2Signer.provider!
+    )
+
+    expect(
+      (await l1User.getBalance()).toString(),
+      'Eth X chain deposit balance'
+    ).to.eq(this.testState.ethXChainDeposit.balance!.toString())
+  }
+}
+
+interface TestState {
+  core: {
+    l1Signer: Signer
+    l2Signer: Signer
+    erc20Bridger: Erc20Bridger
+    ethBridger: EthBridger
+  }
+  ethTransfer: {
+    address?: string
+    balance?: BigNumber
+  }
+  tokenDeploy: {
+    address?: string
+    name?: string
+    symbol?: string
+  }
+  weth: {
+    user?: Wallet
+    balance?: BigNumber
+  }
+  wethXChainDeposit: {
+    user?: Wallet
+    balance?: BigNumber
+  }
+  ethXChainDeposit: {
+    user?: Wallet
+    balance?: BigNumber
+  }
+  wethXChainWithdrawal: {
+    message?: IL2ToL1MessageWriter
+    user?: Wallet
+    balance?: BigNumber
+  }
+  ethXChainWithdrawal: {
+    message?: IL2ToL1MessageWriter
+    user?: Wallet
+    balance?: BigNumber
+  }
+}
+
+class MultiTest extends MigrationTest {
+  public readonly tests: MigrationTest[]
+  constructor(
+    testState: TestState,
+    testConstructors: (new (testState: TestState) => MigrationTest)[]
+  ) {
+    super(testState)
+    this.tests = testConstructors.map(tc => new tc(testState))
+  }
+
+  public async initialise(): Promise<void> {
+    for (const test of this.tests) {
+      console.log(`Initialising ${test.name} test`)
+      await test.initialise()
+      console.log(`Initialising ${test.name} test completed`)
+    }
+    console.log('Tests initialised, ready for migration.')
+  }
+
+  public async finalise(): Promise<void> {
+    console.log('Migration complete testing finalising tests')
+    for (const test of this.tests) {
+      console.log(`Finalising ${test.name} test`)
+      await test.finalise()
+      console.log(`Finalised ${test.name} test`)
+    }
+  }
+}
+
+describe('Migration tests', async () => {
+  const initialiseTestState = async (): Promise<TestState> => {
+    const { l2Signer, l1Signer, erc20Bridger, ethBridger } = await testSetup()
+    await fundL1(l1Signer, parseEther('20'))
+    await fundL2(l2Signer, parseEther('20'))
+
+    return {
+      core: {
+        l2Signer,
+        l1Signer,
+        erc20Bridger,
+        ethBridger,
+      },
+      ethTransfer: {},
+      tokenDeploy: {},
+      weth: {},
+      wethXChainDeposit: {},
+      wethXChainWithdrawal: {},
+      ethXChainWithdrawal: {},
+      ethXChainDeposit: {},
```

Comment:
instead of having many empty things in the test state, we could instead just have `core` be part of `TestState`
then, we have the `initialise` function return a `context` object which is used as a param into the `finalize` function